firebase/src/core/data/gateways/env-gateway.ts

```diff
@@ -20,7 +20,10 @@ export class EnvGateway {
 
   /** Returns `true` if running in local environment, i.e, under Firebase Emulators. */
   get isLocalDevelopment(): boolean {
-    return process.env["FUNCTIONS_EMULATOR"] === "true";
+    const isProduction = process.env["NODE_ENV"] === "production";
+    const isCi = process.env["CI"] === "true";
+
+    return !isProduction && !isCi;
```

Comment:
Updated to use `NODE_ENV` (set when running in Google Cloud environment) and `CI` (set when running in any CI machine) instead `FUNCTIONS_EMULATOR` env variable. This way we can infer if we are running in Google Cloud or in a CI machine without "injecting" extra environment variables in the scripts